pyproject.toml

```diff
@@ -9,23 +9,19 @@ repository = "https://github.com/predict-idlab/powershap"
 keywords = ["feature selection", "shap", "data-science", "machine learning"]
 
 [tool.poetry.dependencies]
-python = ">=3.7.1,<3.10"
-numpy = "~1.20"  # To allow python 3.7.x
-pandas = "~1.3"  # To allow python 3.7.x
+python = ">=3.8,<3.10"
 sklearn = "^0.0"
 catboost = "^1.0.5"
 statsmodels = "^0.13.2"
-shap = "^0.40.0"
+numpy = "^1.23.1"
+pandas = "^1.4.3"
+shap = "^0.41.0"
```

Comment:
Could these changes be a little more flexible in terms of backwards compatibility?
E.g. 
```
numpy ~=1.21
pandas ~=1.4.0
shap ~=0.40.0
```

This keeps them backwards compatible while installing security fixes as and when needed.

Reply:
The easiest example of a conflict that a hard cap such as this raises would be that of `numba`: 

https://github.com/numba/numba/issues/8242

The planned release is in Jan'23, so any environments that use numba optimizations(fairly common, I'd assume) might not be able to use powershap.

Reply:
Sure! Didn't keep this in mind :see_no_evil:

Reply:
Is the caret requirement not more suitable than the tilde requirement in your example?
https://python-poetry.org/docs/dependency-specification/#caret-requirements

Reply:
> Isn't the caret requirement more suitable than the tilde requirement in your example? https://python-poetry.org/docs/dependency-specification/#caret-requirements

Gotcha! Wasn't aware of poetry best practices and the caret option seems more suitable. Thanks for pointing it out.

Reply:
Awesome! Thanks for pushing a release candidate. This works out.

Reply:
No idea.. Just changed it to scikit-learn without any real impact